devicetypes/fibargroup/fibaro-walli-dimmer-switch.src/fibaro-walli-dimmer-switch.groovy

```diff
@@ -0,0 +1,523 @@
+		result << response(["delay 3000", meterGet(scale: 2).format()])
+	}
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+	log.debug "v3 Meter report: "+cmd
+	handleMeterReport(cmd)
+}
+
+def handleMeterReport(cmd) {
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kWh")
+		} else if (cmd.scale == 1) {
+			createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kVAh")
+		} else if (cmd.scale == 2) {
+			createEvent(name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W")
+		}
+	}
+}
+
+def configure() {
+	log.debug "configure()"
+	def result = []
+
+	result << response(encap(meterGet(scale: 0)))
+	result << response(encap(meterGet(scale: 2)))
+}
+
+def on() {
+	encapSequence([
+			zwave.basicV1.basicSet(value: 0xFF),
+			zwave.switchMultilevelV1.switchMultilevelGet(),
+	], 1000)
+}
+
+def off() {
+	encapSequence([
+			zwave.basicV1.basicSet(value: 0x00),
+			zwave.switchMultilevelV1.switchMultilevelGet(),
+	], 1000)
+}
+
+def setLevel(level, rate = null) {
+	if(level > 99) level = 99
+	encapSequence([
+			zwave.basicV1.basicSet(value: level),
+			zwave.switchMultilevelV1.switchMultilevelGet()
+	], 1000)
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	log.debug "ping() called"
+	refresh()
+}
+
+def refresh() {
+	log.debug "refresh()"
+
+	encapSequence([
+			zwave.switchMultilevelV1.switchMultilevelGet(),
+			meterGet(scale: 0),
+			meterGet(scale: 2),
+	], 1000)
+}
+
+def reset() {
+	encapSequence([
+			meterReset(),
+			meterGet(scale: 0)
+	])
+}
+
+def meterGet(scale) {
+	zwave.meterV2.meterGet(scale)
+}
+
+def meterReset() {
+	zwave.meterV2.meterReset()
+}
+
+private encapSequence(cmds, Integer delay=250) {
+	delayBetween(cmds.collect{ encap(it) }, delay)
+}
+
+private encap(physicalgraph.zwave.Command cmd) {
+	if (zwaveInfo?.zw?.contains("s")) {
+		secEncap(cmd)
+	} else {
+		log.debug "no encapsulation supported for command: $cmd"
+		cmd.format()
+	}
+}
+
+private secEncap(physicalgraph.zwave.Command cmd) {
+	log.debug "encapsulating command using Secure Encapsulation, command: $cmd"
+	zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private getParameterMap() {[
+	[
+		name: "LED frame – colour when ON", key: "ledFrame–ColourWhenOn", type: "enum",
+		parameterNumber: 11, size: 1, defaultValue: 1,
+		values: [
+			0: "LED disabled",
+			1: "White",
+			2: "Red",
+			3: "Green",
+			4: "Blue",
+			5: "Yellow",
+			6: "Cyan",
+			7: "Magenta",
+			8: "colour changes smoothly depending on measured power",
+			9: "colour changes in steps depending on measured power"
+		],
+		description: "This parameter defines the LED colour when thedevice is ON. When set to 8 or 9, LED frame colour will change depending on he measured power and parameter 10. Other colours are set permanently and do not depend on power consumption."
```

Comment:
same typos here as in the other PR, and I'd also have the same recommendation about removing the "this parameter..." wording

Reply:
ok, it will be fixed (all details were taken from wave alliance official product data)

Reply:
Fixed, @greens please let me know if it looks better now.